src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -164,6 +169,40 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id,
     return S_OK;
   }
 
+  if (debug_logging_enabled) {
+    Debug("ModuleLoadFinished: ", module_id, " ", module_info.assembly.name,
+            " AppDomain ", module_info.assembly.app_domain_id, " ",
+            module_info.assembly.app_domain_name);
+  }
+
+  AppDomainID app_domain_id = module_info.assembly.app_domain_id;
+
+  // Identify the AppDomain ID of mscorlib which will be the Shared Domain
+  // because mscorlib is always a domain-neutral assembly
+  if (!mscorlib_module_loaded && module_info.assembly.name == "mscorlib"_W) {
+    mscorlib_module_loaded = true;
+    mscorlib_app_domain_id = app_domain_id;
+    return S_OK;
+  }
+
+  // Identify the AppDomain ID of the managed profiler entrypoint
+  if (module_info.assembly.name == "Datadog.Trace.ClrProfiler.Managed"_W) {
```

Comment:
Would saving the module id for use in logging be reasonable? @lucaspimentel would this help in future log parsing and troubleshooting?

Reply:
Sure thing. Can you think of any places that I should log it immediately?

Reply:
Following up on this, I think it would be smart to save the module id for the managed profiler and the runtime's corlib. Do you think we need to log it anywhere besides the line I've already introduced that always runs when we've loaded a valid module?